src/hooks/use-expression-state.ts

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react'
+import Validator from '@/lib/validator'
+import Calculator from '@/lib/calculator'
+
+const ZERO = '0'
+const INFINITY = 'Infinity'
+const INIFINITY_TEXT = '오류'
+
+export const useExpressionState = () => {
+  const [expression, setExpression] = useState(ZERO)
```

Comment:
expression의 타입을 string으로 하면 너무 넓은 것 같아요. 
화면에 표시될 expression의 타입을 최대한 좁힌다면, number 이거나 '오류' 라는 문자열만 있을 것 같네요.
그리고 숫자는 숫자형으로 다루는 게 버그 방지를 위해 좋다고 생각해요. 이리 저리 형 변환을 하다보면 버그가 일어날 수 있을 것 같아서요. 🐞  

```suggestion
const INIFINITY_TEXT = '오류'

export const useExpressionState = () => {
  const [expression, setExpression] = useState<number | INIFINITY_TEXT>(0)
```

Reply:
expression이 계산기에 표시될 상태를 표현하려면 연산자도 들어가야 할텐데 이러면 string이 되어야하지 않을까요??

Reply:
https://github.com/next-step/react-calculator/assets/59413128/b22faf9c-2c96-4013-87ea-fa53f3554e53

앗. 저는 apple ios, mac 에 내장된 계산기
앱의 UX를 중심으로 설명드렸어요.😅

Reply:
아하 여기서는 연산자가 마지막으로 누른것으로 설정되는 UX군요! 
원석님의 의도 완전히 이해했습니다..!